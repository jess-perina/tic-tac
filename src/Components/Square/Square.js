import React from 'react';

function Square(props) {
  return (
    <button type="button" className="square cyan-text text-darken-4" onClick={props.onClick}>
      {props.value}
    </button>
  );
}

export default Square;
