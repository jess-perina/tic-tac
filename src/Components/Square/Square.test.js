import React from 'react'
import { shallow, mount } from 'enzyme'
import Square from './Square'

describe('Square Component', () => {
  it('renders as expected', () => {
    expect(shallow(<Square />)).toMatchSnapshot()
  })

  it('displays the correct value', () => {
    const wrapper = mount(<Square value={4} />)
    const value = wrapper.find('button').render().text()
    expect(value).toBe('4')
  })

  it('calls onClick event when square is clicked', () => {
    const onClick = jest.fn()
    const wrapper = shallow(<Square onClick={onClick} />)
    wrapper.find('button').simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
