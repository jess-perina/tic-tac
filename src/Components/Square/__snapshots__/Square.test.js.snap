// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Component renders as expected 1`] = `ShallowWrapper {}`;
